src/LazyTwigMiddleware.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Slim Framework (http://slimframework.com)
+ *
+ * @license   https://github.com/slimphp/Twig-View/blob/master/LICENSE.md (MIT License)
+ */
+
+declare(strict_types=1);
+
+namespace Slim\Views;
+
+use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use RuntimeException;
+use Slim\App;
+use Slim\Interfaces\RouteParserInterface;
+
+class LazyTwigMiddleware extends TwigMiddleware
+{
+    /**
+     * @var ContainerInterface
+     */
+    protected $container;
+
+    /**
+     * @var string
+     */
+    protected $containerKey;
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function create(App $app, string $containerKey = 'view'): parent
+    {
+        $container = $app->getContainer();
+        if ($container === null) {
+            throw new RuntimeException('The app does not have a container.');
+        }
+
+        return new self(
+            $app->getRouteCollector()->getRouteParser(),
+            $container,
+            $containerKey,
+            $app->getBasePath()
+        );
+    }
+
+    /**
+     * @param RouteParserInterface $routeParser
+     * @param ContainerInterface   $container
+     * @param string               $containerKey
+     * @param string               $basePath
+     */
+    public function __construct(
+        RouteParserInterface $routeParser,
+        ContainerInterface $container,
+        string $containerKey = 'view',
+        string $basePath = ''
+    ) {
+        $this->routeParser = $routeParser;
+        $this->container = $container;
+        $this->containerKey = $containerKey;
+        $this->basePath = $basePath;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $this->twig = $this->container->get($this->containerKey);
```

Comment:
`$this->container->get(...)` may throw an exception when dependency not found. It would be best to use `$this->container->has($containerKey)` to ensure that the dependency exist first.

Reply:
Good catch, will fix!

Reply:
> It should be documented in the PR for reference. Thank you.

So I just write it here?

Reply:
Done

Reply:
If someone creates the instance directly, which means by using the constructior, then this would not be checked before.

Reply:
Done